TCAT/Controllers/ActionOnboardViewController.swift

```diff
@@ -42,67 +51,141 @@ class ActionOnboardViewController: UIViewController, CLLocationManagerDelegate {
-        view.addSubview(title)
+        title.textAlignment = .center
+        title.snp.makeConstraints { (make) in
+            make.bottom.equalToSuperview().offset(-244)
+            make.centerX.equalToSuperview()
+            make.width.equalToSuperview()
+            make.height.equalTo(50)
+        }
         
-        let description = UITextView()
+        description.font = UIFont(name: FontNames.SanFrancisco.Regular, size: 14)
+        description.textColor = descriptionColor
         description.text = getDescription()
-        description.center = view.center
-        description.sizeToFit()
-        description.frame.origin.y = title.frame.maxX + spacing
-        view.addSubview(description)
+        description.textAlignment = .center
+        description.snp.makeConstraints { (make) in
+            make.top.equalTo(title.snp.bottom).offset(12)
+            make.width.equalTo(311.5)
+            make.height.equalTo(80)
+            make.centerX.equalToSuperview()
+        }
+        description.isEditable = false
         
-        let button = UIButton()
-        button.titleLabel?.text = getButtonText()
-        button.frame = CGRect(x: 0, y: 0, width: 100, height: 40)
-        button.backgroundColor = .buttonColor
+        button.setTitle(getButtonText(), for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.titleLabel?.font = UIFont(name: FontNames.SanFrancisco.Medium, size: 16)!
+        button.backgroundColor = mainColor
         button.layer.cornerRadius = 4
-        button.center = view.center
-        button.frame.origin.y = view.bounds.height - 100 - button.frame.height
+        self.button.snp.makeConstraints { (make) in
+            make.bottom.equalToSuperview().offset(-64)
+            make.centerX.equalToSuperview()
+            make.height.equalTo(44)
+        }
+        setButtonConstraints()
+        
+        secondButton.setTitle("Don't Allow", for: .normal)
+        secondButton.setTitleColor(mainColor, for: .normal)
+        secondButton.titleLabel?.font = UIFont(name: FontNames.SanFrancisco.Medium, size: 16)
+        secondButton.backgroundColor = .clear
+        secondButton.snp.makeConstraints { (make) in
+            make.top.equalTo(button.snp.bottom).offset(12)
+            make.width.equalTo(100)
+            make.centerX.equalToSuperview()
+        }
+        secondButton.addTarget(self, action: #selector(dismissOnboarding), for: .touchUpInside)
+        
         if getAction() != nil {
             button.addTarget(self, action: getAction()!, for: .touchUpInside)
         }
-        view.addSubview(button)
         
     }
     
     func getTitle() -> String {
         switch type {
-            case .locationServices: return "Location Services"
-            case .welcome: return "Welcome!"
-            default: return ""
+        case .locationServices: return "Location Services"
+        case .welcome: return "Welcome!"
+        default: return ""
```

Comment:
do u have another case for type besides locationServices or welcome? if not, `default` is not necessary since it'll never be case